geomstats/geometry/positive_lower_triangular_matrices.py

```diff
@@ -0,0 +1,359 @@
+        mat2 = gs.einsum("...ij,...kj->...ik", base_point, tanget_vec)
+        return mat1 + mat2
+
+    @staticmethod
+    def inverse_differential_gram(tanget_vec, base_point):
+        """Compute inverse differential of gram map
+
+        Parameters
+        ----------
+        tangent_vec : array_like, shape=[..., n, n]
+            tanget vector at gram(base_point)
+            Symmetric Matrix
+        base_point : array_like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        inverse_differential_gram : array-like, shape=[..., n, n]
+            Inverse differential of gram.
+            Lower triangular matrix
+        """
+        inv_base_point = gs.linalg.inv(base_point)
+        inv_transpose_base_point = Matrices.transpose(inv_base_point)
+        aux = Matrices.to_lower_triangular_diagonal_scaled(
+            Matrices.mul(inv_base_point, tanget_vec, inv_transpose_base_point)
+        )
+        inverse_differential_gram = Matrices.mul(base_point, aux)
+        return inverse_differential_gram
+
+
+class CholeskyMetric(RiemannianMetric):
+    """Class for the cholesky metric on the cholesky space."""
+
+    def __init__(self, n):
+        """Build the CholeskyMetric
+
+        Parameters
+        ----------
+        n : int
+            Integer representing the shape of the matrices: n x n.
+
+
+        References
+        ----------
+        .. [TP2019] . "Riemannian Geometry of Symmetric
+        Positive Definite Matrices Via Cholesky Decomposition"
+        SIAM journal on Matrix Analysis and Applications , 2019.
+         https://arxiv.org/abs/1908.09326
+        """
+        dim = int(n * (n + 1) / 2)
+        super(CholeskyMetric, self).__init__(
+            dim=dim, signature=(dim, 0), default_point_type="matrix"
+        )
+        self.n = n
+
+    @staticmethod
+    def diag_inner_product(tangent_vec_a, tangent_vec_b, base_point):
+        """Compute the inner product using only diagonal elements.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        tangent_vec_b : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+        base_point : array-like, shape=[..., n, n]
+            Base point.
+
+        Returns
+        -------
+        ip_diagonal : array-like, shape=[...]
+            Inner-product.
+        """
+        inv_sqrt_diagonal = gs.power(Matrices.diagonal(base_point), -2)
+        tangent_vec_a_diagonal = Matrices.diagonal(tangent_vec_a)
+        tangent_vec_b_diagonal = Matrices.diagonal(tangent_vec_b)
+        prod = tangent_vec_a_diagonal * tangent_vec_b_diagonal * inv_sqrt_diagonal
+        ip_diagonal = gs.sum(prod, axis=-1)
+        return ip_diagonal
+
+    @staticmethod
+    def strictly_lower_inner_product(tangent_vec_a, tangent_vec_b):
+        """Compute the inner product using only strictly lower triangular elements.
```

Comment:
To be more precise: compute a term of the inner product, since it is not (yet) the inner product here, same for the description of the Returns

Reply:
nope. It is inner product!. (it satisfies symmetric, non-degenerate, positive bi-linear conditions )

Reply:
Right, it is an inner-product, but not *the* inner-product on the space (which comes on line 246). Looks good like this.